lib/record_reader/json_record_reader.go

```diff
@@ -0,0 +1,118 @@
+package record_reader
+
+import (
+	"bufio"
+	"io"
+)
+
+const expectedJsonRecordSize = 2048
+
+// NewJson returns a record reader that parses Apex-flavor JSON
+// and converts it into bytes of JSONLines format in its records.
+//
+// Note this is not a generic JSON parser- we cannot parse the JSON
+// both because it'd be slower, but especially because it's not really possible
+// to parse JSON like `[{}, {}]` incrementally. So we read lines
+// and figure out when we are between objects.
+//
```

Comment:
jq's stream format offers a possible solution if we find the JSON returned from Salesforce to not be so consistent.  

```
$ echo '[ {
   "x": 1
}, {
        "y": 2' | gojq -n --stream 'fromstream(1|truncate_stream(inputs))'
{
  "x": 1
}
```

Reply:
Hmm, interesting- the incremental parsing requires the custom parser, which jq/gojq seem to have. That would work ('it'd be slower, since it would require parsing, but I suspect it wouldn't be slow enough to notice).